src/NuGet.Core/NuGet.Build.Tasks/RestoreTaskEx.cs

```diff
@@ -146,6 +151,12 @@ public override bool Execute()
                         UseShellExecute = false,
                         WorkingDirectory = Environment.CurrentDirectory,
                     };
+                    
+                    // Add environment variable to generate a restore binlog
+                    if (!string.IsNullOrWhiteSpace(BinlogPath))
+                    {
+                        process.StartInfo.EnvironmentVariables.Add("RESTORE_TASK_BINLOG_PARAMETERS", BinlogPath);
```

Comment:
From a precedence order, env vars tend to win *over* cmd parameters. 

I'd suggest doing the same thing here, check whether that env var has a value before setting it to something new.

Reply:
That ProcessStartInfo object is freshly created and I wouldn't expect it to contain any EnvironmentVariable items in it.

Reply:
Env vars from the current process are inherited on first access of the property.

Reply:
Thanks, I applied your suggestion.